projects/cardiovascular-patient-app/src/app/last-report-bp/last-report-bp.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component } from '@angular/core';
+import { groupBy, mapValues } from 'lodash-es';
+import { DataLayerManagerService, formatDate, formatTime } from 'ngx-charts-on-fhir';
+import { map } from 'rxjs';
+
+interface LastReportedBPdata {
+  systolic: { date: string; value: number };
+  diastolic: { date: string; value: number };
+}
+
+@Component({
+  selector: 'last-report-bp',
+  templateUrl: './last-report-bp.component.html',
+  styleUrls: ['./last-report-bp.component.css'],
+})
+export class LastReportBPComponent {
+  lastReportedBPdata?: LastReportedBPdata;
+
+  constructor(private layerManager: DataLayerManagerService) { }
+
+  ngOnInit(): void {
+    this.layerManager.allLayers$
+      .pipe(
+        map((layers) => {
+          const bloodPressureLayer = layers.find((layer) => layer.name === 'Blood Pressure')?.datasets;
+          const groupedData = groupBy(bloodPressureLayer, (data) => data?.label?.startsWith('Systolic') ? 'systolic' : 'diastolic');
+          const mostRecentData = mapValues(groupedData, (data) => data[data.length - 1].data.at(-1));
```

Comment:
`data[data.length - 1]` is taking an arbitrary dataset from the group. This happens to work for patient 30049 because the dataset with the most recent measurement has a higher index in the datasets array. If you want this to work for every patient, the datasets must be sorted first. Please add unit tests for this.